openedx/features/course_experience/url_helpers.py

```diff
@@ -4,35 +4,139 @@
 Centralizdd in openedx/features/course_experience instead of lms/djangoapps/courseware
 because the Studio course outline may need these utilities.
 """
+from datetime import datetime
+from typing import Optional, Tuple
+
 import six
 from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.http import HttpRequest
 from django.urls import reverse
+from opaque_keys.edx.keys import CourseKey, UsageKey
 from six.moves.urllib.parse import urlencode
 
+from lms.djangoapps.courseware.toggles import courseware_mfe_is_active
+from openedx.core.djangoapps.content.learning_sequences.api import get_course_outline, get_user_course_outline
 from xmodule.modulestore.django import modulestore
 from xmodule.modulestore.search import navigation_index, path_to_location
 
+User = get_user_model()
+
 
-def get_legacy_courseware_url(course_key, usage_key, request=None):
+def get_courseware_url(
+        usage_key: UsageKey,
+        request: Optional[HttpRequest] = None,
+) -> str:
```

Comment:
In this PR, we don't actually use this function anywhere. My [next PR](https://github.com/edx/edx-platform/pull/26816/commits/7736941b8f0a45a1bdaa12370e6c34fe5f395f7c) will update the `/jump_to/` endpoint to use this function, though.